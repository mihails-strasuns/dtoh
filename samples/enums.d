module enums;

enum A { one = 42 }
enum B { one = 1, two = A.one }
enum C { a, b, c }

extern(C) __gshared B var1 = B.two;
extern(C) __gshared A var2;

extern(C) B foo (B x) { return x; }
extern(C) void baz (C);
