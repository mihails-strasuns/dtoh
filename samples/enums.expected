#include <stdint.h>

// Used enum definitions:

enum A
{
    A_one = 42
};
enum B
{
    B_one = 1,
    B_two = 42
};
enum C
{
    C_a = 0,
    C_b = 1,
    C_c = 2
};

// Struct forward declaration to resolve cycles:



// Function pointer types:



// Used struct definitions:



// Variable and function declarations:

enum B var1;
enum A var2;
enum B foo(enum B);
void baz(enum C);
